microprofile-lra/src/main/java/org/jboss/eap/qe/microprofile/lra/LRAParticipantWithFaultTolerance1.java

```diff
@@ -0,0 +1,195 @@
+        try (Client client = ClientBuilder.newClient()) {
+            client.target(uriInfo.getBaseUri() + "/" + LRAParticipantWithFaultTolerance2.PATH)
+                    .request().get();
+        }
+
+        // Cause timeout of this method so exception is thrown and LRA is compensated
+        Thread.sleep(Duration.ofSeconds(10).toMillis());
+
+        return Response.ok("Method with @Timeout ended successfully.").build();
+    }
+
+    /**
+     * If 2 or more calls from last 10 calls fail then circuit is open.
+     * If circuit is open and 1 call is successful (after 1000 ms) delay then circuit is closed again.
+     */
+    @LRA(value = LRA.Type.REQUIRED)
+    @GET
+    @Path(CIRCUIT_BREAKER_PATH)
+    @CircuitBreaker(requestVolumeThreshold = 10, failureRatio = 0.2, delay = 1000, successThreshold = 10)
+    public Response workWithCircuitBreaker(@HeaderParam(LRA.LRA_HTTP_CONTEXT_HEADER) URI lraId,
+            @HeaderParam(LRA.LRA_HTTP_RECOVERY_HEADER) URI recoveryId,
+            @QueryParam(LRAParticipantWithFaultTolerance1.FAIL_LRA) boolean failLRA) throws Exception {
+        LOGGER.infof("Executing action of LRAParticipantWithFaultTolerance1 enlisted in LRA %s " +
+                "that was assigned %s participant Id. Retried: %s", lraId, recoveryId, retryCount.incrementAndGet());
+
+        lraResult.setLraId(lraId);
+        lraResult.setRecoveryId(recoveryId);
+
+        // call Participant 2 to propagate the LRA
+        try (Client client = ClientBuilder.newClient()) {
+            client.target(uriInfo.getBaseUri() + "/" + LRAParticipantWithFaultTolerance2.PATH)
+                    .request().get();
+        }
+
+        if (failLRA) {
+            throw new Exception("Throwing exception to fail participant until circuit is open.");
+        }
+
+        return Response.ok("Method with @CircuitBreaker ended successfully.").build();
+    }
+
+    @LRA(value = LRA.Type.REQUIRED)
+    @GET
+    @Path(FALLBACK_PATH)
+    @Fallback(fallbackMethod = "fallback")
+    public Response workWithFallback(@HeaderParam(LRA.LRA_HTTP_CONTEXT_HEADER) URI lraId,
+            @HeaderParam(LRA.LRA_HTTP_RECOVERY_HEADER) URI recoveryId,
+            @QueryParam(LRAParticipantWithFaultTolerance1.FAIL_LRA) boolean failLRA) throws Exception {
+
+        LOGGER.infof("Executing action of LRAParticipantWithFaultTolerance1 enlisted in LRA %s " +
+                "that was assigned %s participant Id. Retried: %s", lraId, recoveryId, retryCount.incrementAndGet());
+
+        lraResult.setLraId(lraId);
+        lraResult.setRecoveryId(recoveryId);
+
+        // call Participant 2 to propagate the LRA
+        try (Client client = ClientBuilder.newClient()) {
+            client.target(uriInfo.getBaseUri() + "/" + LRAParticipantWithFaultTolerance2.PATH)
+                    .request().get();
+        }
+        if (failLRA) {
+            throw new Exception("Throwing exception to fail participant until circuit is open.");
```

Comment:
How is circuit breaker related here? Shouldn't the message be about FT Fallback?